### ISSUE TITLE
Use "reference_id" colum (item table) to keep an array data

### ISSUE THREAD

**Issue by developez** [amsgames/laravel-shop#43]
Hello

I am dealing with you great package, I want to propose to keep more data in the item table when you don't use a class. At this moment you can keep the sku, price and a id key of a certain class. But if you need to keep an array of ids for example (in my case, array of ids of two classes) you need to modifiy the table, I would to reuse the reference_id column that is null when you create a item without associated class, like this:

```
        $item_data = array();
        $item_data["myclass1"] = $oferta_tipo_id;
        $item_data["myclass2"] = $oferta_fecha_id;
        $item_data["sku"] = "...";
        $item_data["price"] = 100;
        $carro->add($item_data);
```

The idea is to retrieve this array when you read the items in the future:

```
foreach($items as $item)
    $item->getObjectAttribute
```

I can prepare a PR in this way if the owner wants.

**Comment by amsgames** [comment]
The item table is not a product table, that is why it doesn't hold more
data in it.

This gives you freedom to create your own custom product tables with the
properties of your needs.

**Comment by developez** [comment]
Yes, I understand. But with "class" and "reference_id" columns you can only
store one id of a class. If you want to store two ids, you must create
another class with these two ids in two properties or something, and save
that class to a table.

I wonder how to get a good way to obtain a general method to store two o
more id using "class" and "reference_id" columns. I want to avoid
personalice the item tabs with more fields.

In my local fork (not yet uploaded) I did modification to achieve this, so
probably it would'nt like everyone. I mean using reference_id to store an
array of ids, by the moment encoding an array in json.
The item table is not a product table, that is why it doesn't hold more
data in it.

This gives you freedom to create your own custom product tables with the
properties of your needs.

**Comment by amostajo** [comment]
That approach sounds good.

**Comment by developez** [comment]
I've seen some ways:

class = "Array"
reference_id = "['class1_id' => 12, ...](json)

class = "Array"
or reference_id = "142, 767, ..."

class = "Class1, Class2, ..."
reference_id = "12, 874, ..."

I need to this because I am selling products with two involved classes.
